src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -401,4 +411,29 @@ private static String createPoolStatsInfo(String title, PoolStats poolStats) {
     }
     return title;
   }
+
+  private static Boolean bypassProxy(String account){
+    String targetHost = account + SNOWFLAKE_DOMAIN_NAME;
+    if(System.getProperty(NON_PROXY_HOSTS) != null){
+      if(isInNonProxyHosts(targetHost))
+        return true;
+    }
+    return false;
+  }
+
+  private static Boolean isInNonProxyHosts(String targetHost){
+    String nonProxyHosts = System.getProperty(NON_PROXY_HOSTS)
+            .replace(".", "\\.")
+            .replace("*", ".*");
+    String[] nonProxyHostsArray = nonProxyHosts.split("\\|");
+    Pattern p;
+    Matcher m;
+    for(String i : nonProxyHostsArray){
+      p = Pattern.compile(i);
+      m = p.matcher(targetHost);
+      if(m.matches())
```

Comment:
Nit, you could always consolidate: `if Pattern.compile(i).matcher(targetHost).matches() ...`